Enhancement camera recognition occlusion (#6051)
cyberbotics/webots: docs/reference/recognition.md

```
Recognition {
  SFFloat  maxRange          100     # [0, inf)
  SFInt32  maxObjects        -1      # {-1, [0, inf)}
  SFBool   occlusion         TRUE    # {TRUE, FALSE}

>   SFInt32  occlusionAccuracy 1       # {1, 2}

Reviewer: I am wondering if we shouldn't also allow a value of 0 for occlusionAccuracy which would allow us to deprecate the occlusion field.

Author: Yes, probably it will be simpler to have a single filed.
Ideally, then it would be better to call is simply `occlusion` but it might cause issues with backwards compatibility.

Reviewer: I agree it would be better. But since this aims at develop, we can afford to break backwards compatibility. I believe very few Recognition nodes had a non-default value for the occlusion field. So, it shouldn't break many simulations.

Author: Ok, so I will apply these changes.

If more objects are visible to the camera, only the `maxObjects` biggest ones (considering pixel size) are recognized.

- The `occlusion` field defines if occlusions between the camera and the object should be checked.
Disabling the occlusion can be useful to allow the camera to see through thin or transparent objects that may hide the object we are interested in, but it can lead to recognized objects that are not really visible to the camera.
Additionally, it will slightly speed up the simulation.

> - The `occlusionAccuracy` fields defines the accuracy of the occlusion computation.

Reviewer: ```suggestion
- The `occlusionAccuracy` field defines the accuracy of the occlusion computation.
```

`-1` means no limit.
If more objects are visible to the camera, only the `maxObjects` biggest ones (considering pixel size) are recognized.

- The `occlusion` field defines if occlusions between the camera and the object should be checked and the accuracy that will be used.
If the `occlusion` field is set to `0`, then the occlusion computation will be disabled.
Disabling the occlusion can be useful to allow the camera to see through thin or transparent objects that may hide the object we are interested in, but it can lead to recognized objects that are not really visible to the camera.
If the `occlusion` field is set to `1`, only the center of the object is taken into account to compute if the object is visible or not.
Otherwise, if the accuracy is set to `2`, the outbound of the object is used to compute if the object is visible.

> Note the lower the `occlusion` field value is, the higher speed up the simulation achieves.

Reviewer: This sentence sounds odd to me, we could simplify it:
```suggestion
Note that increasing the `occlusion` field value decreases the simulation speed.
```
